packages/dashboard/src/components/common/LogoImg.tsx

```diff
@@ -0,0 +1,88 @@
+import { Box } from "@mantine/core";
+import type { MantineSize, MantineSizes } from "@mantine/core";
+import { useSavedColorScheme } from "src/hooks";
+
+const sizes: MantineSizes = {
+  xs: 30,
+  sm: 42,
+  md: 58,
+  lg: 78,
+  xl: 102
+};
+
+const colorVariants = {
+  normal: {
+    path1: "#5e464d",
+    path2: "#fff",
+    path3: "#fff",
+    path4: "#3fe0c5"
+  },
+  dimmed: {
+    light: {
+      path1: "#e6e0d5",
+      path2: "#f0ede8",
+      path3: "#f0ede8",
+      path4: "#e1dbd0"
+    },
+    dark: {
+      path1: "#282022",
+      path2: "#30282a",
+      path3: "#30282a",
+      path4: "#291f22"
+    }
+  }
+};
+
+type LogoImgProps = {
+  size?: MantineSize | number;
+  variant?: keyof typeof colorVariants;
+};
+
+function LogoImg({ size, variant }: LogoImgProps): JSX.Element {
+  const [colorScheme] = useSavedColorScheme();
+
+  if (typeof size !== "number") {
+    size = sizes[size || "sm"];
+  }
+
+  let colors = colorVariants.normal;
+  if (variant === "dimmed") {
+    colors = colorVariants.dimmed[colorScheme];
+  } else if (colorVariants.hasOwnProperty(variant as string)) {
+    colors = colorVariants[variant!];
+  }
```

Comment:
Looks like this block is causing some trouble for me... maybe re-write this so it's just `const colors = ...`?

Reply:
ah the problem is what happens if `colorScheme` isn't defined... since this was my first load, I had nothing in local storage, so I didn't have any "saved color scheme"... thus `colorVariants.dimmed[colorScheme]` was coming back undefined.

also, how's this work with respect to system color scheme? I'd expect the default to match my system light/dark mode.

Reply:
This is a bug indeed, the first load would fail because of this. I've fixed it [here](https://github.com/trufflesuite/truffle/pull/5241/commits/3fecc13ddde63bcac230cb6ffd3b4bb371f99a21).

It respects system color scheme until you put something in local storage (on toggle button click). The priority is:
1. Local storage, if defined
2. System, if detected (using [Window.matchMedia api](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia))
3. Fallback to light